codesign/test/utils/test_logger.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:logging/logging.dart';
+
+class TestLogger implements Logger {
```

Comment:
I don't think you need this (reusing the existing logger should be fine):

```dart
final List<LogRecord> records = <LogRecord>[];

setUp(() {
  records.clear();
  log.onRecord.listen((LogRecord record) => records.add(record));
});
```

You might need to move your `bin/codesign.dart` into a `lib/src/logging.dart` file so all code has access to the `log` instance

Reply:
The snippet I pasted here is from app_dart (push_gold_status_to_github_test.dart), so you don't need to create a dedicated fake for it

Reply:
Very True! Thanks a ton for explaining!